python/fedml/api/modules/storage.py

```diff
@@ -67,18 +70,24 @@ def upload(data_path, api_key, name, description, tag_list, service, show_progre
 
         file_name = name
 
-    if not to_upload_path:
+    if not to_upload_path and not byte_data_flag:
         return FedMLResponse(code=ResponseCode.FAILURE, message=message)
 
     #TODO(bhargav191098) - Better done on the backend. Remove and pass file_name once completed on backend.
     dest_path = os.path.join(user_id, file_name)
-    file_size = os.path.getsize(to_upload_path)
+    max_chunk_size = 20 * 1024 * 1024
+
+    if byte_data_flag:
+        file_size = sum(len(chunk) for chunk in get_chunks_from_byte_data(byte_data, max_chunk_size))
 
-    file_uploaded_url, message = _upload_multipart(api_key, dest_path, to_upload_path, show_progress,
+    else:
+        file_size = os.path.getsize(to_upload_path)
+
+    file_uploaded_url, message = _upload_multipart(api_key, dest_path, file_size, max_chunk_size, to_upload_path, show_progress,
                                                        out_progress_to_err,
-                                                       progress_desc, metadata)
+                                                       progress_desc, metadata, byte_data_flag, byte_data)
```

Comment:
I think instead of adding so many if-else conditions, we should simplify this a bit by adding sub functions. Instead of having a manyt if-else statements, we can break down the logic into smaller, more manageable functions that each handle a specific scenario and switch the functions out depending on type.

Reply:
Makes sense. This has been addressed with the latest commit.